package/kernel/mac80211/patches/ath/558-ath9k-only-mask-use_eeprom-on-of-noeeprom.patch

```diff
@@ -0,0 +1,75 @@
+ath9k: Avoid OF no-EEPROM quirks without qca,no-eeprom
+
+ath9k_of_init() function[0] was initially written on the assumption that
+if someone had an explicit ath9k OF node that "there must be something
+wrong, why would someone add an OF node if everything is fine"[1]
+(Quoting Martin Blumenstingl)
+
+"it turns out it's not that simple. with your requirements I'm now aware
+of two use-cases where the current code in ath9k_of_init() doesn't work
+without modifications"[1]
+
+The "your requirements" Martin speaks of is the result of the fact that I
+have a device (PowerCloud Systems CR5000) that uses the EEPROM for
+caldata and firmware but for which the MAC address is take from the MTD
+"art" partition[2], or more succinctly:
+
+"some cards come with a physical EEPROM chip so "qca,no-eeprom" should not
```

Comment:
For clarification (I think it's fine to just leave it a physical EEPROM chip though):

Since the AR9300 series (AR9380, AR9382, ... and later cards) the wifi-chip itself has a small One Time Programmable memory area (OTP) that can hold the calibration data in a "diff - like" compressed format.
This is the case for the wifi cards in the Airtight C-60 and some OEM minipcie cards found in laptops.
(In your case the ath9k driver logic handles these card like that they have an eeprom, so the 
qca,no-eeprom can affect these as well).